mailbox/backup/src/test/java/org/apache/james/mailbox/backup/FlagExtraFieldTest.java

```diff
@@ -0,0 +1,302 @@
+    class GetLocalFileData {
+
+        @Test
+        void getLocalFileDataDataShouldThrowWhenNoValue() {
+            FlagsExtraField testee = new FlagsExtraField();
+
+            assertThatThrownBy(() -> testee.getLocalFileDataData())
+                .isInstanceOf(RuntimeException.class);
+        }
+
+        @Test
+        void getLocalFileDataDataShouldReturnByteArraysOfSystemFlagSet() {
+            Flags flags = new Flags();
+            flags.add(Flags.Flag.SEEN);
+            FlagsExtraField testee = new FlagsExtraField(flags);
+
+            assertThat(testee.getLocalFileDataData())
+                .isEqualTo("\\SEEN".getBytes(Charsets.UTF_8));
+        }
+
+        @Test
+        void getLocalFileDataDataShouldReturnByteArrayOfUserFlagSet() {
+            Flags flags = new Flags("myFlags");
+            FlagsExtraField testee = new FlagsExtraField(flags);
+
+            assertThat(testee.getLocalFileDataData())
+                .isEqualTo("myFlags".getBytes(Charsets.UTF_8));
+        }
+
+        @Test
+        void getLocalFileDataDataShouldReturnByteArrayOfSystemAndUserFlagSet() {
+            Flags flags = new Flags("myFlags");
+            flags.add(Flags.Flag.ANSWERED);
+            flags.add(Flags.Flag.SEEN);
+            FlagsExtraField testee = new FlagsExtraField(flags);
+
+            assertThat(testee.getLocalFileDataData())
+                .isEqualTo("\\ANSWERED,\\SEEN,myFlags".getBytes(Charsets.UTF_8));
+        }
+    }
+
+    @Nested
+    class GetCentralDirectoryData {
+
+        @Test
+        void getCentralDirectoryDataShouldThrowWhenNoValue() {
+            FlagsExtraField testee = new FlagsExtraField();
+
+            assertThatThrownBy(() -> testee.getCentralDirectoryData())
+                .isInstanceOf(RuntimeException.class);
+        }
+
+        @Test
+        void getCentralDirectoryDataShouldReturnByteArraysOfSystemFlagSet() {
+            Flags flags = new Flags();
+            flags.add(Flags.Flag.SEEN);
+            FlagsExtraField testee = new FlagsExtraField(flags);
+
+            assertThat(testee.getCentralDirectoryData())
+                .isEqualTo("\\SEEN".getBytes(Charsets.UTF_8));
+        }
+
+        @Test
+        void getCentralDirectoryDataShouldReturnByteArrayOfUserFlagSet() {
+            Flags flags = new Flags("myFlags");
+            FlagsExtraField testee = new FlagsExtraField(flags);
+
+            assertThat(testee.getCentralDirectoryData())
+                .isEqualTo("myFlags".getBytes(Charsets.UTF_8));
+        }
+
+        @Test
+        void getCentralDirectoryDataShouldReturnByteArrayOfSystemAndUserFlagSet() {
+            Flags flags = new Flags("myFlags");
+            flags.add(Flags.Flag.ANSWERED);
+            flags.add(Flags.Flag.SEEN);
+            FlagsExtraField testee = new FlagsExtraField(flags);
+
+            assertThat(testee.getCentralDirectoryData())
+                .isEqualTo("\\ANSWERED,\\SEEN,myFlags".getBytes(Charsets.UTF_8));
+        }
+    }
```

Comment:
what if you have a coma into a userflag ?

Reply:
Right but we can use % as delimiter as flags can not contain it.

Reply:
UserFlag accept any String, then what delimiter to avoid conflict with user flag?

Reply:
are you sure UserFlag should accept any String ?
Did you read IMAP RFC ?